Assets/HoloToolkit/Input/Scripts/GestureManager.cs

```diff
@@ -60,183 +70,418 @@ public partial class GestureManager : Singleton<GestureManager>
+        /// Valid if a manipulation gesture is in progress.
         /// </summary>
         public Vector3 ManipulationOffset { get; private set; }
 
         /// <summary>
-        /// The world space position of the hand being used for the current manipulation gesture.  Not valid
-        /// if a manipulation gesture is not in progress.
+        /// The world space position of manipulation source being used for the current manipulation gesture.
+        /// Valid if a manipulation gesture is in progress.
         /// </summary>
-        public Vector3 ManipulationHandPosition
+        public Vector3 ManipulationPosition
         {
             get
             {
-                Vector3 handPosition = Vector3.zero;
-                currentHandState.properties.location.TryGetPosition(out handPosition);
-                return handPosition;
+                Vector3 position;
+                if (!currentInteractionSourceState.properties.location.TryGetPosition(out position))
+                {
+                    position = Vector3.zero;
+                }
+                return position;
             }
         }
 
+        /// <summary>
+        /// InteractionSourceDetected tracks the interaction detected state.
+        /// Returns true if the list of tracked interactions is not empty.
+        /// </summary>
+        public bool InteractionSourceDetected
+        {
+            get { return trackedInteractionSource.Count > 0; }
+        }
+
+        #endregion
+
+#if UNITY_EDITOR || UNITY_STANDALONE
+        /// <summary>
+        /// Key to press that will select the currently focused object.
+        /// </summary>
+        public KeyCode keyboardSelectKey = KeyCode.Space;
+
+        /// <summary>
+        /// Enumerated Mouse Buttons.
+        /// </summary>
+        public enum MouseButton
+        {
+            Left = 0,
+            Right = 1,
+            Middle = 2
+        }
+
+        /// <summary>
+        /// Mouse button to press that will select the current focused object.
+        /// </summary>
+        public MouseButton MouseSelectButton = MouseButton.Right;
+#endif
+
         private GestureRecognizer gestureRecognizer;
-        // We use a separate manipulation recognizer here because the tap gesture recognizer cancels
-        // capturing gestures whenever the GazeManager focus changes, which is not the behavior
-        // we want for manipulation
+
+        /// <summary> We use a separate manipulation recognizer here because the tap gesture recognizer cancels
+        /// capturing gestures whenever the GazeManager focus changes, which is not the behavior
+        /// we want for manipulation
+        /// </summary>
         private GestureRecognizer manipulationRecognizer;
 
-        private bool hasRecognitionStarted = false;
+        private InteractionSourceState currentInteractionSourceState;
+
+        private HashSet<uint> trackedInteractionSource = new HashSet<uint>();
```

Comment:
@darax Ahh, I remember what they're used for. UX. that way people can trigger the cursor to change when a source is detected or pressed.